src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -658,6 +658,30 @@ private CorInfoType asCorInfoType(TypeDesc type)
 
             if (type.IsValueType)
             {
+                if (_compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.X86)
+                {
+                    LayoutInt elementSize = type.GetElementSize();
+
+#if READYTORUN
+                    if (elementSize.IsIndeterminate)
+                    {
+                        throw new RequiresRuntimeJitException(type);
+                    }
+#endif
+                    if (elementSize.AsInt == 4)
+                    {
+                        var normalizedCategory = _compilation.TypeSystemContext.NormalizedCategoryFor4ByteStructOnX86(type);
+                        if (normalizedCategory != type.Category)
+                        {
+                            if (NeedsTypeLayoutCheck(type))
```

Comment:
Can you put this under `#if READYTORUN` to avoid build break in runtimelab?

Reply:
Sure. Done.